core/src/main/java/nu/yona/server/messaging/service/MessageService.java

```diff
@@ -209,4 +209,14 @@ public void removeMessagesFromUser(MessageDestinationDTO destination, UUID sentB
 		destinationEntity.removeMessagesFromUser(sentByUserAnonymizedID);
 		MessageDestination.getRepository().save(destinationEntity);
 	}
+
+	public static boolean getIsDirectMessage(MessageDTO message)
+	{
+		if (message instanceof BuddyConnectRequestMessageDTO)
```

Comment:
This approach is a little tricky. Over time, we might add more types of direct messages or, even worse, send some type of message through both message boxes. Let's for now stick with it, as it makes the overall code flow mor readable.